core/store/src/contract.rs

```diff
@@ -50,41 +48,38 @@ impl ContractsTracker {
     }
 
     fn call(&mut self, code_hash: CodeHash) {
-        self.uncommitted_calls.insert(code_hash);
+        self.contract_calls.insert(code_hash);
     }
 
     fn deploy(&mut self, code: ContractCode) {
         self.uncommitted_deploys.insert((*code.hash()).into(), code);
     }
 
-    fn clear_uncommitted(&mut self) {
+    /// Rollsback the uncommitted deployments.
+    fn rollback_deploys(&mut self) {
         self.uncommitted_deploys.clear();
-        self.uncommitted_calls.clear();
     }
 
-    fn commit(&mut self) {
+    /// Commits the uncommitted deployments by moving them to the set of committed deployments and clearing the uncommitted list.
+    fn commit_deploys(&mut self) {
         let deploys = std::mem::take(&mut self.uncommitted_deploys);
         for (code_hash, contract) in deploys.into_iter() {
             self.committed_deploys.insert(code_hash, contract);
         }
-
-        let calls = std::mem::take(&mut self.uncommitted_calls);
-        for code_hash in calls.into_iter() {
-            self.committed_calls.insert(code_hash);
-        }
     }
 
-    fn get_committed(&self) -> ContractStorageResult {
+    /// Finalizes this tracker and returns the calls and committed deployments.
+    fn finalize(self) -> ContractStorageResult {
         ContractStorageResult {
-            contract_calls: self.committed_calls.iter().cloned().collect(),
-            contract_deploys: self.committed_deploys.keys().cloned().collect(),
+            contract_calls: self.contract_calls.into_iter().collect(),
```

Comment:
```suggestion
            contract_calls: mem::take(&mut self.contract_calls),
```

FWIW this reminds me of an example from the offsite where we were talking about type-state guarantees. Converting from `BTreeSet` to a `Vec` here drops type-level guarantees that a) values are unique; and b) that they are sorted. Why not just keep `BTreeSet` in `ContractStorageResult`?